DisCatSharp.Docs/articles/misc/hosting.md

```diff
@@ -22,6 +22,119 @@ Any modern hardware will work just fine, new or used.
+```
+
+### Cloud Hosted Code
+
+- If you have your code hosted with git or simply are able to download it, get the source code and `cd` into the directory. Example of a git repository on GitHub:
+```sh
+git clone https://github.com/username/repo.git
+cd repo
+```
+- Add your configuration file then build the project.
+
+### Local Source Code
+
+- If you do not have your code hosted with git let's move the code inside the VM by following the given steps: 
+
+- Enable developer options on your phone, find out [here](https://developer.android.com/studio/debug/dev-options) and also enable USB debugging which you can do by going inside `Settings > Developer options > USB debugging` or where ever your phone's is located.
+
+- Find the option `Default USB configuration` and choose `Transferring files`. Disconnect your phone from the usb and reconnect it again.
+
+> [!NOTE]
+> A prompt should come up saying a new device is connected to the PC which is the phone. Go to `This PC` on Windows and double click on your device.
+
+- Double click on `Phone` or what it is named, then simply copy your project folder from your pc and paste it in there. If you want to keep it in a custom path, remember the path to the project.
+
+>[!NOTE]
+> Transferring of files will take a bit and be sure to not interrupt the process by shutting down any devices or disconnecting.
+
+- First, we setup Termux storage run, `termux-setup-storage` and then it'll prompt you to give Termux access to your files and folders. Then run, `ls ~/storage/shared` to make sure you have access.
+
+>[!WARNING]
+> If you do not give Termux access to your files and folders, nothing will work going from here on.
+
+- Now, exit out of Debian and into normal Termux and make a new environment variable which leads us to the VM's root directory. Run, `export PROOTDISTROFS=/data/data/com.termux/files/usr/var/lib/proot-distro/installed-rootfs` and run `source $HOME/.bashrc`.
+
+>[!NOTE]
+>  If you do not have a `.bashrc` in your `$HOME` just run, `nano $HOME/.bashrc`, save and exit. Then run, the source command again.
+
+- Now, let's ove over the source code from our phone to the VM. Simply, run, `mv projectname/ $PROOTDISTROFS/debian/root/`.
+
+>[!NOTE]
+> Double, check for file path's as you may accidentally move the wrong files around.
+
+- Now, let's login back to Debian `proot-distro login debian` and check if it's here. To check, run `ls` and if you need `projectname` in the list then it's there. From here.
+
+- Add your configuration file then build the project.
+
+>[!NOTE]
+> If you have folders such as `bin/` and `obj/` from your prior builds on Windows, delete those by running, `rm -rf bin/ obj/` as without doing so, will result in an error. 
+
+## Profit
+
+Bot should be working fine, given you follow appropriate steps. For support or any inquires you can join the DisCatSharp [Discord](https://discord.com/invite/2HWta4GXus).
```

Comment:
I don't know if we want to support this method officially.
Termux is a cool suggestion, but in any case a real "vps" would be better.

Reply:
> I don't know if we want to support this method officially. Termux is a cool suggestion, but in any case a real "vps" would be better.

Well, I just wanted to add a cheaper alternative as to a VPS or something since, if you have an old phone lying around it's just way cheaper and convenient for small or local scaled bots.